### PR TITLE
Top Level WAF Swagger Changes

### PR DESCRIPTION
**Pull request by venk8**
Link To Previous PR:
https://github.com/Azure/azure-rest-api-specs/pull/6851

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by anton-evseev — CHANGES_REQUESTED** [review]
This PR changes specs in 2019-06-01 but it has to target 2019-08-01

**Comment by venkatsvpr** [comment]
@number213  I have done the same. Please check.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: Shouldn't there also be `Enabled`?

**Review by anton-evseev — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: No. By default every rule is enabled. when user wants to disable an rule, a Disabled entry has to be added. The "Disabled" enum is introduced so as to keep the config in parity with AFD Waf.

Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: In `WafPolicyCreateOrUpdate.json` state is set as `Enabled` - if the customer can pass that value and it's going to be accepted, it should be added to Swagger.

Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: Also, there's a cotradiction. In property description it says `Defaults to Disabled if not specified.` but you say `By default every rule is enabled`

Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: Talked in Teams, resolved, description will be updated to avoid confusion

Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: @number213, As discussed  i will update the description. 

Just to summarize: for us all the rules  are enabled. If an entry is there to be overriden it means that user wants to disable it. Thats why we dont have a enabled entry. The reason we have the disabled is for the config to look similar to AFD Waf.

**Comment by number213** [comment]
@majastrz please review this from ARM's side

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json
Comment (majastrz) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: >state [](start = 9, length = 5)

It looks like you are changing a name of a property in a stable API version, which is a breaking change. You should introduce this in a new API version.

**Review by majastrz — CHANGES_REQUESTED** [review]
There seem to be breaking changes in your PR. I added a comment about this. Please take a look.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json
Comment (number213) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: 2019-08-01 is the new API version

**Comment by number213** [comment]
@majastrz this is a branch for new API version that's not merged into `master` yet. Could you please approve this PR if there are no other concerns?

**Review by anton-evseev — APPROVED** [review]

**Review by majastrz — APPROVED** [review]

**Comment by number213** [comment]
@zikalino could you please review and merge?

**Comment by number213** [comment]
Per call with @venkatsvpr there may be some changes so adding "do not merge" tag for now

**Review by anton-evseev — CHANGES_REQUESTED** [review]
Please update `WafPolicyCreateOrUpdate.json`. If you send properties from the current example, you'd get an error:

```
PS C:\Users\v-anevse> ARMClient.exe PUT /subscriptions/947d47b4-7883-4bb9-9d85-c5e8e2f572ce/resourceGroups/v-anevse/providers/Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies/test-policy?api-version=2019-08-01 "{\`"location\`":\`"WestUs\`",\`"properties\`":{\`"customRules\`":[{\`"name\`":\`"Rule1\`",\`"priority\`":1,\`"ruleType\`":\`"MatchRule\`",\`"action\`":\`"Block\`",\`"matchConditions\`":[{\`"matchVariables\`":[{\`"variableName\`":\`"RemoteAddr\`",\`"selector\`":null}],\`"operator\`":\`"IPMatch\`",\`"matchValues\`":[\`"192.168.1.0/24\`",\`"10.0.0.0/24\`"]}]},{\`"name\`":\`"Rule2\`",\`"priority\`":2,\`"ruleType\`":\`"MatchRule\`",\`"matchConditions\`":[{\`"matchVariables\`":[{\`"variableName\`":\`"RemoteAddr\`",\`"selector\`":null}],\`"operator\`":\`"IPMatch\`",\`"matchValues\`":[\`"192.168.1.0/24\`"]},{\`"matchVariables\`":[{\`"variableName\`":\`"RequestHeaders\`",\`"selector\`":\`"UserAgent\`"}],\`"operator\`":\`"Contains\`",\`"matchValues\`":[\`"Windows\`"]}],\`"action\`":\`"Block\`"}]}}"

{
  "error": {
    "code": "ApplicationGatewayFirewallManagedRuleSetsNoValidCrsRuleSetsAttached",
    "message": "Firewall policy /subscriptions/947d47b4-7883-4bb9-9d85-c5e8e2f572ce/resourceGroups/v-anevse/providers/Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies/test-policy does not have any valid Core Rule Set attached to it.",
    "details": []
  }
}
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/examples/WafListAllPolicies.json
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/examples/WafListAllPolicies.json]: Shouldn't this response also have `maxRequestBodySizeInKb` and other new props?

**Review by anton-evseev — CHANGES_REQUESTED** [review]
LGTM except one example

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/examples/WafListAllPolicies.json
Comment (venkatsvpr) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/examples/WafListAllPolicies.json]: yes. Added

**Review by anton-evseev — APPROVED** [review]

**Comment by number213** [comment]
@tjprescott assigning this to you since Catherine Zhu said you're one of those who would cover for Chinese team during their long holiday. Could you please merge this PR?

File: specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json
Comment (sergey-shandar) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/webapplicationfirewall.json]: Add new properties at the end.

**Pull request merged**
